ruby/src/IceRuby/Types.cpp

```diff
@@ -3169,7 +3149,6 @@ IceRuby::initTypes(VALUE iceModule)
     rb_define_module_function(iceModule, "__defineDictionary", CAST_METHOD(IceRuby_defineDictionary), 3);
     rb_define_module_function(iceModule, "__declareProxy", CAST_METHOD(IceRuby_declareProxy), 1);
     rb_define_module_function(iceModule, "__declareClass", CAST_METHOD(IceRuby_declareClass), 1);
```

Comment:
I guess we have declare and define class for forward-declaration purposes? Is this really necessary in Ruby?

Reply:
I think it is with the current code, if the forward declare type is used in an operation we pass the type to the method that defines the operation.